components/infra-proxy-service/integration_test/clients_test.go

```diff
@@ -9,16 +9,94 @@ import (
 )
 
 func TestGetClients(t *testing.T) {
+	// Pre-populated that has been added by scripts
+	// Validating the clients search based on these records.
 	// rpc GetClients (request.Clients) returns (response.Clients)
 	ctx := context.Background()
 	req := &request.Clients{
 		ServerId: autoDeployedChefServerID,
 		OrgId:    autoDeployedChefOrganizationID,
 	}
-	res, err := infraProxy.GetClients(ctx, req)
-	assert.NoError(t, err)
-	assert.NotNil(t, res)
-	assert.Equal(t, 3, len(res.GetClients()))
+
+	t.Run("Clients list without a search params", func(t *testing.T) {
+		// Default per_page 1000 is set in backend
+		perPage := int32(1000)
+		res, err := infraProxy.GetClients(ctx, req)
+		assert.NoError(t, err)
+		assert.NotNil(t, res)
+		assert.Equal(t, int(res.Page), 0)
```

Comment:
These tests are too specific to the sample data loaded. Let's make sure that we modify them when we are going to add edit/create apis.